pkg/api/api_default_service.go

```diff
@@ -1167,3 +1173,100 @@ func (cfm *CfmApiService) RootGet(ctx context.Context) (openapi.ImplResponse, er
 	}
 	return openapi.Response(http.StatusOK, response), nil
 }
+
+type Device struct {
+	Hostname string `json:"hostname"`
+	Address  string `json:"address"`
+	Port     int    `json:"port"`
+	Type     string `json:"type"`
+}
+
+// DiscoverDevices -
+func (cfm *CfmApiService) DiscoverDevices(ctx context.Context, type_ string) (openapi.ImplResponse, error) {
+	if type_ != "cma" && type_ != "cxl-host" {
+		err := common.RequestError{
+			StatusCode: http.StatusBadRequest,
+			Err:        fmt.Errorf("invalid type parameter"),
+		}
+		return formatErrorResp(ctx, &err)
+	}
```

Comment:
I believe my YAML changes above should eliminate the need for this code.  You should be able to delete lines 1186-1192.  Let the openapi generated code handle this check, via the `enum`.

Reply:
Thank you, Scott, but I have checked, the generated code doesn't include this check.

Reply:
It will if you add the `enum` to the `discoveredDevices` schema as I suggested

Reply:
I have added the enum to discoveredDevices but haven't seen the code in generated code. 
![image](https://github.com/user-attachments/assets/5e0329b0-a7ce-4eed-948f-5141aba5f805)

Reply:
It won't be there.  It's probably be in the "model" code for `DiscoverDevices`.  Try it (now or when you're doing the webui work).  If you pass in anything other than `blade` or `cxl-host`, the request will be rejected.

Reply:
Nevermind.  The enums aren't working like I thought.  I'll resolve this conversation.